lib/chain/ChainClient.ts

```diff
@@ -1,12 +1,107 @@
-import BaseClient from '../BaseClient';
+import { EventEmitter } from 'events';
+import { ChainTypes } from '../consts/Types';
+import RpcClient, { RpcConfig } from '../RpcClient';
+import ChainInterface from './ChainInterface';
 
-// TODO: listen to transactions sent to specifc addresses
+type Info = {
+  version: number;
+  protocolversion: number;
+  blocks: number;
+  timeoffset: number;
+  connections: number;
+  proxy: string;
+  difficulty: number;
+  testnet: boolean;
+  relayfee: number;
+};
 
-/**
- * A generic interface that can be used for multiple chain nodes and implementations
- */
-interface ChainClient extends BaseClient {
-  sendRawTransaction(rawTransaction: string): Promise<any>;
+type Block = {
+  hash: string;
+  confirmations: number;
+  strippedsize: number;
+  size: number;
+  weight: number;
+  height: number;
+  version: number;
+  versionHex: string;
+  merkleroot: string;
+  tx: string[];
+  time: number;
+  nonce: number;
+  bits: string;
+  difficulty: number;
+  previousblockhash: string;
+  nextblockhash: string;
+};
+
+interface ChainEvents {
+  on(event: 'error', listener: (error: string) => void): this;
+  on(event: 'transaction', listener: (transactionHex: string) => void): this;
+  emit(event: 'error', error: string): boolean;
+  emit(event: 'transaction', transactionHex: string): boolean;
+}
+
+class ChainClient extends EventEmitter implements ChainInterface, ChainEvents {
```

Comment:
We will have to think about the naming again once we add a chain client which is not similar to BTCD but we can leave it as is for now.